torchsparse/nn/modules/conv.py

```diff
@@ -193,3 +203,47 @@ def forward(self, inputs: SparseTensor):
         outputs = outputs.view(batch_size, *self.bev_shape, -1)
         outputs = outputs.permute(0, 3, 1, 2).contiguous()
         return outputs
+
+
+class ToBEVHeightCompression(nn.Module):
+    def __init__(self,
+                 channels: int,
+                 shape,
+                 offset: list = [0, 0, 0],
+                 dim: int = 1,
+                 bias: bool = False) -> None:
+        super().__init__()
+        self.channels = channels
+        self.offset = torch.cuda.IntTensor([list(offset) + [0]])
+        self.dim = dim
+        self.bev_dims = [i for i in range(3) if i != self.dim]
+        self.bev_shape = shape[self.bev_dims].int()
+        self.shape = shape.int()
+
+    def __repr__(self):
+        return 'ToBEVHeightCompression(channels=%d)' % (self.channels)
+
+    def forward(self, inputs: SparseTensor):
+        coords, feats, stride = inputs.C, inputs.F, inputs.s
+        # [b, x, y, z]
+        coords = (coords - self.offset).t()[[3] + self.bev_dims +
+                                            [self.dim]].long()
+        shape = self.shape[self.bev_dims + [self.dim]]
+        if not isinstance(stride, int):
+            dim = self.dim
+            stride = stride[:, self.bev_dims + [self.dim]]
+            stride = stride.t()
+        coords[1:] = (coords[1:] // stride).long()
+        coords[-1] = torch.clamp(coords[-1], 0, shape[-1] - 1)
+        indices = coords[0] * int(shape.prod()) + coords[1] * int(
+            shape[1:].prod()) + coords[2] * int(shape[2]) + coords[3]
+        batch_size = coords[0].max().item() + 1
+        outputs = torch.cuda.sparse.FloatTensor(
```

Comment:
We might use `torch.sparse_coo_tensor` (see https://pytorch.org/docs/stable/sparse.html#sparse-tensor-functions for more details) instead of `torch.cuda.sparse.FloatTensor ` to support different devices.

Reply:
Implemented. Pending experiments to see whether results will be influenced.